airbyte-integrations/connectors/destination-postgres/src/test-integration/java/io/airbyte/integrations/destination/postgres/typing_deduping/PostgresTypingDedupingTest.java

```diff
@@ -107,4 +156,43 @@ protected List<JsonNode> dumpRawTableRecords(String streamNamespace, String stre
     return super.dumpRawTableRecords(streamNamespace, streamName.toLowerCase());
   }
 
+  @Test
+  public void testVarcharLimitOver64K() throws Exception {
+    final ConfiguredAirbyteCatalog catalog = new ConfiguredAirbyteCatalog().withStreams(List.of(
+        new ConfiguredAirbyteStream()
+            .withSyncMode(SyncMode.FULL_REFRESH)
+            .withDestinationSyncMode(DestinationSyncMode.OVERWRITE)
+            .withStream(new AirbyteStream()
+                            .withNamespace(streamNamespace)
+                            .withName(streamName)
+                            .withJsonSchema(SCHEMA))));
+
+
+
+    final AirbyteMessage message = new AirbyteMessage();
+    final String largeString = generateBigString();
+    final TestMessage testMessage =
```

Comment:
Not sure I see the benefit of the TestMessage class? I feel like it would be clearer to either directly build a map here, or even explicitly build an objectnode:
```java
Map<String, Object> data = ImmutableMap.of(
  "id1", 1,
  "id2", 200,
  "updated_at", "2021-01-01T00:00:00Z",
  "name", largeString);
message....withData(Jsons.jsonNode(data));
```
```java
ObjectNode data = Jsons.deserialize(
  """
  {
    "id1": 1,
    "id2": 200,
    "updated_at": "2021-01-01T00:00:00Z"
  }
  """);
data.put("name", largeString);
message....withData(data);
```

specifically - instead of needing the reader to scroll up + grok the additionalProperties / `JsonAnyGetter` stuff, we could do something completely local to this method, with less verbosity (i.e. no need to call `withAdditionalProperty` 4 times), using abstractions that the reader should already be familiar with

Reply:
I guess. was originally trying to create a fully typed object and ended up with additionalProperties (being lazy). I'll put it out into actual test